webrender/src/platform/unix/font.rs

```diff
@@ -53,7 +53,56 @@ unsafe impl Send for FontContext {}
 
 extern "C" {
     fn FT_GlyphSlot_Embolden(slot: FT_GlyphSlot);
-    fn FT_GlyphSlot_Oblique(slot: FT_GlyphSlot);
+}
+
+const OBLIQUE_SKEW_FACTOR: f32 = 0.2;
+
+fn get_skew_bounds(bottom: i32, top: i32) -> (f32, f32) {
+    let skew_min = ((bottom as f32 + 0.5) * OBLIQUE_SKEW_FACTOR).floor();
+    let skew_max = ((top as f32 - 0.5) * OBLIQUE_SKEW_FACTOR).ceil();
+    (skew_min, skew_max)
+}
+
+fn skew_bitmap(bitmap: &mut Vec<u8>, width: &mut usize, height: usize, left: &mut i32, top: i32) {
+    let stride = *width * 4;
+    // Calculate the skewed horizontal offsets of the bottom and top of the glyph.
+    let (skew_min, skew_max) = get_skew_bounds(top - height as i32, top);
+    // Allocate enough extra width for the min/max skew offsets.
+    let skew_width = *width + (skew_max - skew_min) as usize;
+    let mut skew_buffer = vec![0u8; skew_width * height * 4];
+    for y in 0 .. height {
+        // Calculate a skew offset at the vertical center of the current row.
+        let offset = (top as f32 - y as f32 - 0.5) * OBLIQUE_SKEW_FACTOR - skew_min;
+        // Get a blend factor in 0..256 constant across all pixels in the row.
+        let blend = (offset.fract() * 256.0) as u32;
+        let mut dest = (y * skew_width + offset.floor() as usize) * 4;
+        let (mut prev_b, mut prev_g, mut prev_r, mut prev_a) = (0u32, 0u32, 0u32, 0u32);
+        for src in bitmap[y * stride .. (y + 1) * stride].chunks(4) {
```

Comment:
This could probably be done using ```zip``` and ```iter_mut``` so that it's not necessary to manually increment the dest buffer index?